### DISCUSSION TITLE
Compilation in windows fails

### DISCUSSION THREAD

**Discussion by HectorFire** [openssl/openssl#21453] [Q&A]
I am trying to compile in windows using the following configuration
perl Configure --api=1.1.0 --debug zlib --with-zlib-include="C:\Program Files\zlib\include" --with-zlib-lib="C:\Program Files\zlib\lib\zlibstatic.lib" VC-WIN64A

I have installed zlib in this path.

When it tries to link with zlib lib the compilation stops a send the following error:
`LINK : fatal error LNK1104: cannot open file 'C:\Program.obj'
Could Not Find C:\Users\h484551\repo\openssl-OpenSSL_1_1_1a\libcrypto-1_1-x64.*
NMAKE : fatal error U1077: 'link' : return code '0x1'`

I think the issue is the space in the path between Program and File because if I move zlib to C:\ directly and replace de configuration the compilation complete succesfully.
perl Configure --api=1.1.0 --debug zlib --with-zlib-include="C:\zlib\include" --with-zlib-lib="C:\zlib\lib\zlibstatic.lib" VC-WIN64A

Do you know how can I fix the space issue?

**Comment by t8m** [comment]
> I am trying to compile in windows using the following configuration perl Configure --api=1.1.0 --debug zlib --with-zlib-include="C:\Program Files\zlib\include" --with-zlib-lib="C:\Program Files\zlib\lib\zlibstatic.lib" VC-WIN64A
> 
> I have installed zlib in this path.
> 
> When it tries to link with zlib lib the compilation stops a send the following error: `LINK : fatal error LNK1104: cannot open file 'C:\Program.obj' Could Not Find C:\Users\h484551\repo\openssl-OpenSSL_1_1_1a\libcrypto-1_1-x64.* NMAKE : fatal error U1077: 'link' : return code '0x1'`
> 
> I think the issue is the space in the path between Program and File because if I move zlib to C:\ directly and replace de configuration the compilation complete succesfully. perl Configure --api=1.1.0 --debug zlib --with-zlib-include="C:\zlib\include" --with-zlib-lib="C:\zlib\lib\zlibstatic.lib" VC-WIN64A
> 
> Do you know how can I fix the space issue?

Windows should still support the shortened names, or don't they anymore? If this problem can be reproduced on recent 3.x releases this can be reported as a bug and it should be eventually fixed. (1.1.1a is old and even for the latest 1.1.1 we do security fixes only).

**Comment by nhorman** [comment]
Could you also just escape the space? i.e.:

Perl Configure --api=1.1.0 --debug zlib --with-zlib-include=\"C:\Program\ Files\zlib\include\" --with-zlib-lib="C:\Program\\ 
 Files\zlib\lib\zlibstatic.lib"
